dspace-server-webapp/src/test/java/org/dspace/app/rest/LoginAsEPersonIT.java

```diff
@@ -0,0 +1,150 @@
+                        .andExpect(jsonPath("$._embedded.eperson",
+                                            EPersonMatcher.matchEPersonOnEmail(eperson.getEmail())));
+
+
+    }
+
+    @Test
+    public void loggedInAsOtherUserNotAUuidInHeaderBadRequestRetrievalTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .param("projection", "full")
+                                     .header("X-On-Behalf-Of", "not-a-uuid"))
+                        .andExpect(status().isBadRequest());
+
+
+    }
+
+    @Test
+    public void loggedInAsOtherUserWrongUuidInHeaderBadRequestRetrievalTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .param("projection", "full")
+                                     .header("X-On-Behalf-Of", UUID.randomUUID()))
+                        .andExpect(status().isBadRequest());
+
+
+    }
+
+    @Test
+    public void loggedInAsOtherUserNoPermissionForbiddenRetrievalTest() throws Exception {
+
+
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .param("projection", "full")
+                                     .header("X-On-Behalf-Of", eperson.getID()))
+                        .andExpect(status().isForbidden());
+
+
+    }
+
+
+    @Test
+    public void loggedInUserPropertyFalseTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        configurationService.setProperty("webui.user.assumelogin", false);
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .header("X-On-Behalf-Of", eperson.getID()))
+                        .andExpect(status().isBadRequest());
+
+        context.turnOffAuthorisationSystem();
+        configurationService.setProperty("webui.user.assumelogin", true);
+        context.restoreAuthSystemState();
+    }
+
+
+    @Test
+    public void loggedInUserOtherAdminTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson testEperson = EPersonBuilder.createEPerson(context).withEmail("loginasuseradmin@test.com").build();
+
+
+        Group adminGroup = groupService.findByName(context, Group.ADMIN);
+        groupService.addMember(context, adminGroup, testEperson);
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .header("X-On-Behalf-Of", testEperson.getID()))
+                        .andExpect(status().isBadRequest());
+
+
+
+
+    }
+
```

Comment:
I would like to have an additional test using the loginAs feature in some core functionalities like
- workspaceitems (to check that the item created with the loginAs are owned by the impersonated user or that the user can search for his items)
- tasks are claimed according to the loginas
- actions reserved to an admin are prevent to the loginAs (for instance trying to delete an archived item after loginas a normal user)